config/s3/config.go

```diff
@@ -104,6 +104,9 @@ func Configure(p *config.Provider) {
 			Type:      "github.com/upbound/provider-aws/apis/s3/v1beta1.Bucket",
 			Extractor: `github.com/crossplane/upjet/pkg/resource.ExtractParamPath("arn",true)`,
 		}
+		r.References["rule.destination.encryption_configuration.replica_kms_key_id"] = config.Reference{
+			Type: "github.com/upbound/provider-aws/apis/kms/v1beta1.Key",
```

Comment:
If I understand correctly, we need to capture the ID of kms.Key resource as the reference value. Without overriding the extractor, we will use the default `ExternalName` extractor. What do you think about adding the ID extractor: 

```suggestion
			Type: "github.com/upbound/provider-aws/apis/kms/v1beta1.Key",
			Extractor: common.PathTerraformIDExtractor,
```

Reply:
in the field description for replica_kms_key_id i can see 

`	// ID (Key ARN or Alias ARN) of the customer managed AWS KMS key stored in AWS Key Management Service (KMS) for the destination bucket.`

Reply:
As I see, the external name of kms.Key is arn of this resource so the default extractor works.